scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -788,7 +789,7 @@ private final Connection getInitialConnection(Connection connection, boolean rem
 			if (added) {
 				return peerConnection;
 			} else {
-				terminateOngoingHandshake(peerConnection, new IllegalStateException("connection store exhausted!"),
+				terminateOngoingHandshake(connection, new IllegalStateException("connection store exhausted!"),
```

Comment:
This was a bug from f61c070 too ?
(I asked to be sure if I well understand)

Should we use `connection` too, several lines above : `connectionStore.remove(peerConnection, removeFromSessionCache);`

Reply:
Yes, good catch! And even here peerConnection ...